src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -199,28 +199,33 @@ private void UpdateSize()
                 if (extraField.Size < sizeof(long))
                     return true;
 
-                long value64 = reader.ReadInt64();
+                // Advancing the stream (by reading from it) is possible only when:
+                // 1. There is an explicit ask to do that (valid files, corresponding boolean flag(s) set to true, #77159).
+                // 2. When the size indicates that all the information is available ("slightly invalid files", #49580).
```

Comment:
If you're going to include issue/PR numbers, please include the full link.  We've already moved code between repos once, and the meaning of such numbers changes when we do.

Reply:
My personal taste is to not include them in product code, and rely on comments, links in tests, and commit comments. But that's from seeing code bases full up with 20 years of bug numbers, and get harder to read for it. But I am not sure that is a consensus view 🙂